src/txdb.cpp

```diff
@@ -264,11 +280,39 @@ bool CCoinsViewDB::BatchWrite(CCoinsMap &mapCoins,
     
     for (CCswNullifiersMap::iterator it = cswNullifies.begin(); it != cswNullifies.end();) {
         const std::pair<uint256, libzendoomc::ScFieldElement>& position = it->first;
-        BatchWriteCswNullifier(batch, position.first, position.second, it->second.entered);
+        BatchWriteCswNullifier(batch, position.first, position.second, (it->second.flags == CCswNullifiersCacheEntry::FRESH));
         CCswNullifiersMap::iterator itOld = it++;
         cswNullifies.erase(itOld);
     }
 
+    // Store cert Data Hashes
+    for (CCertDataHashMap::iterator it = certDataHashes.begin(); it != certDataHashes.end();) {
+        const std::pair<uint256, int>& position = it->first;
+        uint256 scId = position.first;
+        int epochId = position.second;
+        std::pair<libzendoomc::ScFieldElement, libzendoomc::ScFieldElement> prevData;
+        libzendoomc::ScFieldElement cumulativeCertData;
+
+        switch(it->second.flags) {
+            case CCertDataHashCacheEntry::Flags::FRESH:
+            case CCertDataHashCacheEntry::Flags::DIRTY:
+                if (GetCertData(scId, epochId - 1, prevData)) {
+                    cumulativeCertData = calculateCumulativeCertDataHash(prevData.first, prevData.second);
```

Comment:
Let's put the comment somewhere, that we store cumulative cert data hash up to the `epochNumber` (so without the current epoch `certDataHash`.